packages/web-components/src/components/va-date-text-input/va-date-text-input.tsx

```diff
@@ -0,0 +1,256 @@
+
+    // Check validity of date if invalid provide message and error state styling
+    if (
+      yearNum < 1900 ||
+      yearNum > 2200 ||
+      monthNum < 1 ||
+      monthNum > 12 ||
+      dayNum < 1 ||
+      dayNum > daysForSelectedMonth.length ||
+      (this.required && !isFullDate(this.value))
+    ) {
+      this.error = 'Please provide a valid date';
+    } else if (this.customValidationBoolean) {
+      this.error = this.customValidationMessage;
+    } else {
+      this.error = '';
+    }
+    this.dateBlur.emit(event);
+  };
+
+  private handleDateChange = (event: Event) => {
+    const target = event.target as HTMLInputElement;
+    let [currentYear, currentMonth, currentDay] = (this.value || '').split('-');
+    if (target.classList.contains('input-month')) {
+      currentMonth = target.value;
+    }
+    if (target.classList.contains('input-day')) {
+      currentDay = target.value;
+    }
+    if (target.classList.contains('input-year')) {
+      currentYear = target.value;
+    }
+
+    this.value = `${currentYear}-${currentMonth ? currentMonth : ''}-${
+      currentDay ? currentDay : ''
+    }`;
+
+    // This event should always fire to allow for validation handling
+    this.dateChange.emit(event);
+  };
+
+  private handleDateKey = (event: KeyboardEvent) => {
+    // Allow 0-9 and then Backspace and Tab to clear data or move to next field
+    const validKeys = [
+      '0',
+      '1',
+      '2',
+      '3',
+      '4',
+      '5',
+      '6',
+      '7',
+      '8',
+      '9',
+      'Backspace',
+      'ArrowRight',
+      'ArrowLeft',
+      'Tab',
+    ];
+    if (validKeys.indexOf(event.key) < 0) {
+      event.preventDefault();
+      return false;
+    }
+  };
+
+  /**
+   * Whether or not an analytics event will be fired.
+   */
+  @Prop() enableAnalytics: boolean;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * input value changes and enableAnalytics is true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  render() {
+    const {
+      required,
+      label,
+      name,
+      error,
+      handleDateBlur,
+      handleDateChange,
+      handleDateKey,
+      value,
+      ariaDescribedby,
+    } = this;
+
+    // Convert string to number to remove leading 0 on values less than 10
+    const [year, month, day] = (value || '').split('-').map(val => val);
+
+    // Setting new attribute to avoid conflicts with only using error attribute
+    // Error attribute should be leveraged for custom error messaging
+    // Fieldset has an implicit aria role of group
+    return (
+      <Host
+        value={value}
+        error={error || this.error}
+        onBlur={handleDateBlur}
+        aria-describedby={ariaDescribedby}
+      >
+        <fieldset aria-label="Input month and day fields as two digit XX and four digit year format XXXX">
```

Comment:
Does the `XX` and `XXXX` need to be read out by screenreaders? Probably something Angela Fowler can answer for us.

Reply:
So this was originally provided to me by our former accessibility specialist Courtney, but @SarahKay8 I'd be interested to hear your opinion on what you think the aria-label should read

Reply:
Hey @pdavies88 so a couple of things to note 

 1. To label a fieldset, use the legend element.
2. Each input field must be individually labelled. It does not show here if they are.
3. If they are individually labelled, your fieldset should say what the date is, for example “Birthday” or “Order date”.
See also for reference 

https://www.w3.org/WAI/tutorials/forms/grouping/#related-fields

Reply:
From @SarahKay8 not sure why it is not showing up here but - Hey [@pdavies88](https://github.com/pdavies88) so a couple of things to note

1. To label a fieldset, use the legend element.
2. Each input field must be individually labelled. It does not show here if they are.
3. If they are individually labelled, your fieldset should say what the date is, for example “Birthday” or “Order date”.
4. See also for reference - https://www.w3.org/WAI/tutorials/forms/grouping/#related-fields